### PR TITLE
Pr/fixing whisper bug and updating build instructions

### PR DESCRIPTION
**Pull request by Amaan1234567**
fixed bugs where it wouldnt compile cpu version when no GPU was detected, updated build instructions to add openvino installation and formatted all files edited with clang format

Checklist(to mark the completed tasks, replace the space inside the brackets with an x):
- [x] Are the variable and function named correctly. Example: ```myNewVariable```
- [x] Are the class and structs named correctly. Example: ```MyNewClass```
- [x] Are the comments clear and concise.
- [x] Is the documentation updated.
- [x] There are no new warnings than before*.
- [x] Are the logs used correctly. Only logs are allowed. No print statements.

*The only reason new warnings may be allowed is if the code that produces the warnings are from third party code that we cannot modify.(Example: Qt, KDE Frameworks, git submodules etc.)

### CODE REVIEW

**Review by RahulVadhyar — CHANGES_REQUESTED** [review]
openvino is showing warning/error in gh actions. I don't think it is getting compiled properly. see that once

File: src/ui/app.cpp [hunk 1]
```diff
@@ -48,8 +48,9 @@ App::App(int argc, char *argv[])
 }
 
 int App::run()
-{
-    //LOG_INFO("app", this->model_api_instance->transcriptionService());
+{   
+    
+    LOG_INFO("app", this->model_api_instance->transcriptionService());
```
Comment (RahulVadhyar) [inline src/ui/app.cpp]: turn this off as it messes with the UI

File: src/model_backend/model_api.hpp [hunk 2]
```diff
@@ -49,10 +48,14 @@ class ModelApi
...
 
 public:
-
     // Constructor with initializer list
-    ModelApi(std::string model = "moondream:1.8b-v2-q5_K_M", std::string transcriptionModel = "ggml-small.en.bin",std::string systemPrompt = "You are a helpful AI assistant, who is good at solving the user's queries be it related to coding, general knowledge or anything else a human assistant is capable of. you also have excellent image recognition and analysis skills due to have multi model capabilites, now following will be the user's queries, analyse them carefully and repsond back to the user properly. ");
-    
+    ModelApi(std::string model = "llava-phi3:3.8b",
+             std::string transcriptionModel = "ggml-small.en.bin",
+             std::string systemPrompt =
+                 "You are a helpful AI assistant, who is good at solving the user's queries be it related to coding, general knowledge or anything else a "
```
Comment (RahulVadhyar) [inline src/model_backend/model_api.hpp:55]: make this const so that its purpose is clear

File: src/model_backend/thirdparty/CMakeLists.txt [hunk 3]
```diff
@@ -1,19 +1,77 @@
...
+#enable_language(CUDA)
+
+#set(CMAKE_CUDA_ARCHITECTURES "60;70;75;80;86" CACHE STRING "CUDA architectures")
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
 set(BUILD_SHARED_LIBS OFF)
-# set(WHISPER_OPENVINO ON)
-set(GGML_CUDA ON)
```
Comment (RahulVadhyar) [inline src/model_backend/thirdparty/CMakeLists.txt]: this is not needed as the whisper.cpp cmake does all this.

**Review by RahulVadhyar — CHANGES_REQUESTED** [review]
The cmake hack is not getting merged.

**Review by RahulVadhyar — APPROVED** [review]

**Comment by RahulVadhyar** [comment]
please squash and merge @shrenisc

**Review by shrenisc — APPROVED** [review]
approved.

**Pull request merged**